### PR TITLE
Support WebStorm 2022.2 EAP

### PR DESCRIPTION
**Pull request by martinbeentjes**
Currently JetBrains is running a closed beta with the new UI design. I would like to have the latest EAP to support the plugin. I do not know if it will work, reading from the docs this should do the trick from the side of IDEA telling it is supported.

### CODE REVIEW

File: src/main/java/dev/blachut/svelte/lang/codeInsight/SvelteComponentCandidatesProvider.kt [hunk 1]
```diff
@@ -33,7 +33,7 @@ class SvelteComponentCandidatesProvider(placeInfo: JSImportPlaceInfo) : JSImport
         return true
     }
 
-    override fun getNames(keyFilter: Predicate<String>): Set<String> {
+    override fun getNames(keyFilter: Predicate<in String>): MutableSet<String> {
```
Comment (tomblachut) [inline src/main/java/dev/blachut/svelte/lang/codeInsight/SvelteComponentCandidatesProvider.kt:36]: Why did you change `Set` into `MutableSet`?

File: build.gradle.kts [hunk 2]
```diff
@@ -76,7 +79,7 @@ val generateSvelteLexer = task<GenerateLexerTask>("generateSvelteLexer") {
 }
 
 tasks {
-    val javaVersion = "11"
+    val javaVersion = "17"
```
Comment (tomblachut) [inline build.gradle.kts:82]: Please don't change more than necessary, let's stay on Java 11 & Kotlin 1.5 for maximum compatibility

File: src/main/java/dev/blachut/svelte/lang/codeInsight/SvelteKitModuleReferenceContributor.kt [hunk 3]
```diff
@@ -65,10 +65,10 @@ class SvelteKitModuleReferenceContributor : JSResolvableModuleReferenceContribut
...
     // based on TypeScriptUtil.createFilterByNodeModuleScope
     private fun createFilterByNodeModuleScope(scope: GlobalSearchScope, context: PsiElement): GlobalSearchScope {
-        var context = context
-        if (context is PsiFile) context = context.originalFile
-        val contextFile = PsiUtilCore.getVirtualFile(context)
-        val index = ProjectFileIndex.getInstance(context.project)
+        var mutableContext = context
+        if (mutableContext is PsiFile) mutableContext = mutableContext.originalFile
+        val contextFile = PsiUtilCore.getVirtualFile(mutableContext)
+        val index = ProjectFileIndex.getInstance(mutableContext.project)
```
Comment (tomblachut) [inline src/main/java/dev/blachut/svelte/lang/codeInsight/SvelteKitModuleReferenceContributor.kt:71]: This change are irrelevant too. It pollutes git history for no tangible gain 😕

**Review by tomblachut — CHANGES_REQUESTED** [review]

File: src/main/java/dev/blachut/svelte/lang/completion/SvelteAttributeValueCompletionProvider.kt [hunk 4]
```diff
@@ -53,7 +53,7 @@ class SvelteAttributeValueCompletionProvider : CompletionProvider<CompletionPara
         result.add("css")
         CSSLanguage.INSTANCE.dialects.forEach {
             if (it.displayName != "JQuery-CSS") {
-                result.add(it.displayName.toLowerCase(Locale.US))
```
Comment (martinbeentjes) [inline src/main/java/dev/blachut/svelte/lang/completion/SvelteAttributeValueCompletionProvider.kt:56]: This was a deprecation warning.

File: src/main/java/dev/blachut/svelte/lang/codeInsight/SvelteComponentCandidatesProvider.kt [hunk 5]
```diff
@@ -33,7 +33,7 @@ class SvelteComponentCandidatesProvider(placeInfo: JSImportPlaceInfo) : JSImport
         return true
     }
 
-    override fun getNames(keyFilter: Predicate<String>): Set<String> {
```
Comment (martinbeentjes) [inline src/main/java/dev/blachut/svelte/lang/codeInsight/SvelteComponentCandidatesProvider.kt:36]: This was necessary to fix the 'getNames overrides nothing' error.

**Comment by martinbeentjes** [comment]
@tomblachut I've fixed the mentioned issues.

**Review by tomblachut — APPROVED** [review]

**Comment by tomblachut** [comment]
@martinbeentjes thanks again for the contribution!

**Pull request merged**